lib/signs_ui_web/channels/signs_channel.ex

```diff
@@ -50,6 +50,27 @@ defmodule SignsUiWeb.SignsChannel do
     end
   end
 
+  def handle_in("changeChelseaBridgeAnnouncements", changes, socket) do
+    case socket_access_level(socket) do
```

Comment:
Some of these access control bits now appear duplicated 3 times in this module — maybe worth an extraction? Like:

```elixir
with_admin_access(socket, fn ->
  {:ok, _new_state} = SignsUi.Config.State.update_chelsea_bridge_announcements(changes)
  username = Guardian.Phoenix.Socket.current_resource(socket)

  Logger.info(
    "chelsea_bridge_announcements_changed: user=#{username} changes=#{inspect(changes)}"
  )
end)
```

This could have a better name to more clearly imply what it does, but this is the best one I could think up at the moment.

Reply:
👍 pushed up this change. looks good.